toml.md

```diff
@@ -61,6 +61,10 @@ Comments may contain any Unicode code points except the following control codes
 that could cause problems during editing or processing: U+0000, and U+000A to
 U+000D.
 
+Comments are not considered part of the table that the document describes, and
+parsers must never be affected by the presence, or contents, of comments within
+the file when generating that table.
+
```

Comment:
Personally I'd use plainer language for this; for example:

> Comments cannot affect parser behaviour and must always be completely ignored by a parser. They are exclusively for human consumption.

Still not 100% happy with that, but best I could think of right now.

"Part of the table that the document describes" in particular seems rather awkward to me. I had to read that twice.

Reply:
That wouldn't work since comment-preserving parsers are allowed to read them and write them back. They wouldn't be able to do that if comments "must always be completely ignored by a parser."

Reply:
Given that this is the first meaningful section of the specification, and the TOML-specific term "table" has not been introduced yet, I do think it's better to we avoid using that term. Besides, I don't think a TOML document describes a table. :)

Here's my attempt (I'd put this before the line on unicode codepoints, since it's less "implementation-detail" and many readers will likely skim past that line):

> Comments should be used to communicate between the human readers of a file. Parsers must not modify keys or values, based on the presence (or contents) of a comment.

I'm fine with mentioning keys and values here since they're hinted at in the example above, and "Key/Value Pair" is literally the next section.

Reply:
> That wouldn't work since comment-preserving parsers are allowed to read them and write them back. They wouldn't be able to do that if comments "must always be completely ignored by a parser."

Ah right; you could tweak it a bit:

> Comments must not affect parser behaviour and the content must be ignored by a parser. They are exclusively for human consumption.

I'm also fine with Pradyun's version (minus the comma after "keys or values"), although I think "affect parser behaviour" is clearer than "modify keys or values".

Reply:
I think @pradyunsg has the right idea here, I'll use his text. I was reluctant to mention what comments were for (since we assumed what they were for, before), so I'm glad that this sentiment was reintroduced. And good point about tables, too. I wanted to draw a distinction between a TOML table and the sort of table-like structure that programming languages have that correspond to TOML tables, but it's true that that isn't the best place in `toml.md` to bring these concepts up. The comma is a little weird, but I think it makes the statement flow more easily. I'll apply my flexible sense of where commas can be, and keep it in place. (What the heck, I use Oxford commas all the time, and I haven't been arrested once!)